doc/rtd/reference/custom_modules/custom_cleaners.rst

```diff
@@ -0,0 +1,28 @@
+.. _custom_cleaners:
+
+Custom Cleaners
+***************
+
+Cloud-init provides the directory :file:`/etc/cloud/clean.d/` for third party
+applications which need additional configuration artifact cleanup from
+the filesystem when the :ref:`cloud-init clean<cli_clean>` command is invoked.
+
+The :command:`clean` operation is typically performed by image creators
+when preparing a golden image for clone and redeployment. The clean command
+removes any cloud-init semaphores, allowing cloud-init to treat the next
+boot of this image as the "first boot". When the image is next booted
+cloud-init will performing all initial configuration based on any valid
```

Comment:
I'm not sure this sentence is needed at all, but `performing` should be `perform`.

Reply:
Removed.